### PR TITLE
interop-testing: support just ServerBuilder types

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java [hunk 1]
```diff
@@ -233,14 +234,17 @@ private void startServer() {
...
         .addStreamTracerFactory(serverStreamTracerFactory);
-    io.grpc.internal.TestingAccessor.setStatsImplementation(
-        builder,
-        new CensusStatsModule(
-            tagger,
-            tagContextBinarySerializer,
-            serverStatsRecorder,
-            GrpcUtil.STOPWATCH_SUPPLIER,
-            true, true, true, false /* real-time metrics */));
+    if (metricsExpected()) {
```
Comment (zhangkun83) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: Even if metrics are not expected, census context propagation may still happen, e.g., in `InProcessTest`. You need to inject fake Census whenever builder is `AbstractServerImplBuilder`, regardless of `metricsExpected()`, and then have the assert:
```java
if (metricsExpected()) {
  assertThat(builder).isInstanceOf(AbstractServerImplBuilder.class).
}
```

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java [hunk 2]
```diff
@@ -1472,6 +1476,7 @@ public void customMetadata() throws Exception {
   @Test(timeout = 10000)
   public void censusContextsPropagated() {
     Assume.assumeTrue("Skip the test because server is not in the same process.", server != null);
+    Assume.assumeTrue(metricsExpected());
```
Comment (zhangkun83) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: Same reason as above. This should be `assumeTrue(serverBuilderIsAbstractServerImplBuilder)`.

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java (hunk 1 above)
Comment (carl-mastrangelo) [reply interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: I changed it to just check the class.  PTAL

Comment (zhangkun83) [reply interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: Please add the assert I suggested. It's much easier to debug this assert failing than metrics not being recorded.

Comment (carl-mastrangelo) [reply interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java]: Done, (sorry, I didn't understand why you wanted it at first).

**Pull request merged**